src/main/java/io/cryostat/rules/RuleProcessor.java

```diff
@@ -232,17 +232,19 @@ private void archiveRuleRecording(ConnectionDescriptor connectionDescriptor, Rul
         targetConnectionManager.executeConnectedTask(
                 connectionDescriptor,
                 connection -> {
-                    IRecordingDescriptor descriptor =
-                            connection.getService().getSnapshotRecording();
-                    try {
-                        recordingArchiveHelper
-                                .saveRecording(connectionDescriptor, descriptor.getName())
-                                .get();
-                    } finally {
-                        connection.getService().close(descriptor);
-                    }
+                    try (connection) {
```

Comment:
I should have edited that commit message to reword it. I had a `cryostat-core` feature branch I was experimenting with where `JFRConnection` held an internal reference count, and calling `connection.close()` would decerement the count. Only when the count reached 0 would the actual underlying connection be closed.

After some more testing and thinking about it I figured this reference counting didn't really help or offer anything that couldn't just be done within the `TargetConnectionManager` on the application side. This try-with-resources still ends up working as originally intended when combined with the later commit in this PR that adds the `executeConnectedTask(connDesc, task, false)` method for the non-cached connections.

Reply:
... on second thought, after explaining that back again, it's wrong. The `try-with-resources` here forces the connection to be closed, even if it was one that was originally pulled from the cache. That isn't what I ended up implementing in the end with the latest version of `executeConnectedTask(, , false)`. That would only have worked with an earlier version of it that always returned a new connection and never reused one from the cache. Since it is possible that the connection used is one from the cache, I have removed the try-with-resources. The TCM will be responsible for cleaning up the connection here, if needed.

Reply:
Sounds good, that makes sense.